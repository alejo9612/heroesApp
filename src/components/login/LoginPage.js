import React, { useContext } from 'react';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const LoginPage = ({history}) => {

    const {dispatch} = useContext(AuthContext)

    const handleLogin=()=>{

        const lastPath = localStorage.getItem('lastPath')||'/';

        //history.push('/'); me redirecciona
        dispatch({
            type: types.login,
            payload: {
                name:'Carnaval'
            }
        });

        history.replace(lastPath);
        //history.replace('/');//me redirecciona y elimina el historial anterior de la page
    }

    return (
        <div className="container mt-5">
            <h1>Login</h1>
            <hr/>

            <button className="btn btn-success"
                    onClick={handleLogin}
            >
                Login
            </button>
        </div>
    )
}
