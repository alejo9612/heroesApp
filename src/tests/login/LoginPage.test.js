import { mount } from "enzyme";
import { AuthContext } from "../../auth/AuthContext";
import { LoginPage } from "../../components/login/LoginPage";
import { types } from "../../types/types";

describe('Se llevarán a cabo las pruebas de <LoginPage/>', () => {

    const history = {
        replace: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            name: 'Carnaval',
            logged: true
        }
    }
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <LoginPage history={history} />
        </AuthContext.Provider>
    );

    test('Debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de realizar el dispatch y la navegación', () => {
        
        wrapper.find('button').prop('onClick')();
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload:{name:'Carnaval'}
        });

        expect(history.replace).toHaveBeenCalledWith('/');

    });

});
