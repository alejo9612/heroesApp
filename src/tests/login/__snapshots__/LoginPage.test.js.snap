// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se llevar√°n a cabo las pruebas de <LoginPage/> Debe de mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
